elixir/apps/domain/lib/domain/resources.ex

```diff
@@ -46,6 +47,26 @@ defmodule Domain.Resources do
     end
   end
 
+  def count_resources_for_gateway(%Gateways.Gateway{} = gateway, %Auth.Subject{} = subject) do
```

Comment:
`with :ok <- Auth.ensure_has_permissions(subject, required_permissions) do` is missing here, we should make sure that the `subject` has permission to access `resources` data in the first place.

Reply:
Gotcha.  I saw some other places that (I think) didn't have it and I knew this would be filtered by the `subject.account.id`, so wasn't sure what if we needed it since it was just a count being returned.  Happy to add it though.

Reply:
@bmanifold it's more about security boundaries (that live in the domain for us) and code consistency, we don't want the auth subject to fetch any data from a domain context if it doesn't have permission to access it.

Reply:
Yep, totally fair and agreed.  I'll add it in.

Reply:
Added the `Auth.ensure_has_permissions` check